File: code-samples/messaging/global-business/get-sync-dlr.js
const MessenteApi = require('messente_api');

const defaultClient = MessenteApi.ApiClient.instance;
const basicAuth = defaultClient.authentications['basicAuth'];
basicAuth.username = 'YOUR_MESSENTE_API_USERNAME';
basicAuth.password = 'YOUR_MESSENTE_API_PASSWORD';

const api = new MessenteApi.DeliveryReportApi();

api.retrieveDeliveryReport('<omnimessage_id>', (error, data) => {
  if (error) {
    console.error(error);
  } else {
    console.log('API called successfully. Returned data: ', data);
  }
});

File: code-samples/messaging/global-business/get-sync-dlr.java
import com.messente.ApiClient;
import com.messente.ApiException;
import com.messente.api.*;
import com.messente.auth.HttpBasicAuth;

public class GetSyncDlr {
    public static void main(String[] args) {
        ApiClient apiClient = new ApiClient();
        DeliveryReportApi apiInstance = new DeliveryReportApi(apiClient);

        HttpBasicAuth basicAuth = (HttpBasicAuth) apiClient.getAuthentication("basicAuth");
        basicAuth.setUsername("YOUR_MESSENTE_API_USERNAME");
        basicAuth.setPassword("YOUR_MESSENTE_API_PASSWORD");

        try {
            Object result = apiInstance.retrieveDeliveryReport("<omnimessage_id>");
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when retrieving delivery report");
            System.err.println(e.getResponseBody());
        }
    }
}
